### PR TITLE
WIP - adds todo ui

### PR DESCRIPTION
**Pull request by narendra-jtc**
## Description
 - added ui components for todo task (wip)
 - functionality of (adding, editing, deleting) tasks remaining
 
 ## Database schema changes
N/A

## Tests
N/A

Screenshot

### CODE REVIEW

File: src/apps/frontend/components/button/button.styles.ts [hunk 1]
```diff
@@ -13,13 +13,16 @@ const styles = {
...
       `,
       disableState: 'cursor-not-allowed bg-primary/80',
       enableState: 'hover:bg-primary/90 cursor-pointer',
     },
     secondary: {
       base: `
+        p-2
+        rounded
+        hover:bg-slate-200
+        bg-white
```
Comment (jjalan) [inline src/apps/frontend/components/button/button.styles.ts:25]: Let's make sure we are not causing regression or undesired experience at other places which uses secondary button.

File: src/apps/frontend/components/textarea/textarea.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+interface TextAreaProps {
+  disabled: boolean;
+  error: string;
+  name: string;
+  placeholder: string;
+  rows: number;
+}
+
+const TextArea: React.FC<TextAreaProps> = ({
```
Comment (jjalan) [inline src/apps/frontend/components/textarea/textarea.tsx:11]: Should this be `Textarea`? If it should be `TextArea`, the file/folder name should be `text_area`.

**Review by jjalan — COMMENTED** [review]
- While creating pull request, kindly keep the [PR template sections](https://github.com/jalantechnologies/rflask-boilerplate/blob/main/.github/pull_request_template.md) as is, and fill them out or put N/A.

- Whenever UI change is done, please add video/images of the UI changes in the PR to help visualize the change.

File: src/apps/frontend/pages/tasks/task-checkbox.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+interface TaskCheckboxProps {
+  taskId: string;
+}
+
+const TaskCheckBox: React.FC<PropsWithChildren<TaskCheckboxProps>> = ({
+  children,
+  taskId,
+}) => (
+  <label htmlFor={taskId} className="flex cursor-pointer select-none">
+    <div className="flex items-center content-center">
```
[the remark this answers is not in the document]
Comment (narendra-jtc) [reply src/apps/frontend/pages/tasks/task-checkbox.tsx]: html is quite forgiving with these kind of usage, but yes its not the best practice, updated with inline element usage to ensure better compatibility and fewer layout issues

File: src/apps/frontend/components/textarea/textarea.tsx (hunk 2 above)
Comment (narendra-jtc) [reply src/apps/frontend/components/textarea/textarea.tsx]: yes , this should be Textarea, updated is PR

File: src/apps/frontend/components/button/index.tsx [hunk 4]
```diff
@@ -5,36 +5,51 @@ import { ButtonKind, ButtonType } from '../../types/button';
...
 }
 
 const Button: React.FC<PropsWithChildren<ButtonProps>> = ({
   children,
   disabled,
+  fullWidth,
   isLoading,
+  onBlur,
   onClick,
+  startIcon,
```
[the remark this answers is not in the document]
Comment (narendra-jtc) [reply src/apps/frontend/components/button/index.tsx]: renamed to better propname -> icon, this way we can add 'position' in future without messing with existing usage

File: src/apps/frontend/components/button/button.styles.ts (hunk 1 above)
Comment (narendra-jtc) [reply src/apps/frontend/components/button/button.styles.ts:25]: yes, checked.

**Comment by jjalan** [comment]
A screenshot does not suffice in this case.

We need to show a video that shows the state of screen for each state - default, loading (when operation is in progress), error (when server encountered error or there is an issue with field validation), success (when operation completed successfully) etc.

Under Test section, we need to detail all the manual tests that you have conducted. This helps reviewer to know what all scenarios you covered. We need to be verbose about this section.

**Comment by bbbneo333** [comment]
@narendra-jtc - Requesting to work on a private fork. This PR is not intended for merge here. Closing this one.

**Pull request closed without merging**